### ISSUE TITLE
set minimum R version and minimum Rust version in `use_extendr()`

### ISSUE THREAD

**Issue by etiennebacher** [extendr/rextendr#375]
Related to https://github.com/etiennebacher/astgrepr/issues/28

Compiling `astgrepr` from source fails on R 4.0.3 with this error:

```
error[E0425]: cannot find value `S4SXP` in this scope
   --> /tmp/RtmpRgdx7u/R.INSTALL1ae525e0ee11/astgrepr/src/vendor/extendr-api/src/lib.rs:574:15
    |
574 |         S4 => S4SXP,
    |               ^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`.
error: could not compile `extendr-api` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...
make: *** [Makevars:19: rust/target/release/libastgrepr.a] Error 101
```

I suppose this is because `extendr` takes advantage of new features in R, so compiling from source won't work for older versions of R. But does this mean users on R 4.0 won't be able to install the binary as well?

More generally, does using `extendr` affect the minimal R version supported? If so, this should be documented somewhere but I didn't find anything on this.

**Comment by CGMossa** [comment]
So I think we have a feature gate for this, so it should work for older versions of R. In theory. Something could have slipped up.

I believe we only don't support the R version that still had 32-bit support, because that's too much for us to handle. But that's an unofficial policy that I'm just speaking of here.

There could be a bug, I'm unsure :( I don't have time to investigate unfortunately. But if you do try to investigate it, I'm sure you'd find a bug somewhere.. I have feature flags in extendr for this.

**Comment by etiennebacher** [comment]
I also don't have time to search for this, especially since I'm not familiar at all with the codebase, but I think a message when initiating `rextendr` (or even automatically update the `DESCRIPTION` with `Depends: R (>= x.y.z)`) would be welcome.

**Comment by etiennebacher** [comment]
Related: https://github.com/extendr/extendr/issues/664

**Comment by Ilia-Kosenkov** [comment]
You might be able to get away by referencing older versions of `extendr`. `{rextendr}` itself depends on R >= 4.0, and should not use features from later versions. https://github.com/extendr/rextendr/blob/c4f5b0cea27f144edc2544bbfed1f35f505c3aa8/DESCRIPTION#L40

We provide support for current, oldrel and devel versions of R (current, previous and future minor versions).

You might also be able to use latest extendr with `bindgen`, but then it will require having heavy `clang` integration in order to generate bindings for the version of language that you use.

**Comment by etiennebacher** [comment]
> You might be able to get away by referencing older versions of extendr.

I think having a document referencing the versions of R that are supported in each version of `extendr` would be very useful. For example a table like:

| `extendr` version | Oldest supported R version |
|-------------------|----------------------------|
| 0.6.0             | 4.0                        |
| 0.7.0             | 4.2                        |

I just put some random R versions here.

> We provide support for current, oldrel and devel versions of R (current, previous and future minor versions).

If you don't want to have this kind of table and only provide support for current, oldrel-1, and devel, then I think this should be reflected in the `DESCRIPTION` when running `rextendr::use_extendr()`.

Another alternative would be to have features for each R version, so that one can have:

```toml
extendr-api = { version = "0.7.0", features = ["R-4.1"], optional = true }
```

(but I don't know if this is hard to implement). Currently, it's impossible to know [which of the features](https://docs.rs/crate/extendr-api/latest/features) blocks some versions of R. I have no idea which approach is the best in this case.

**Comment by JosiahParry** [comment]
Yes. I agree. I suggest we set the minimum supported version of R to 4.2 by default. We can accomplish this like so: 

```r
use_r_version <- function(version) {
  usethis::use_package("R", "Depends", version)
}
```

then in `use_extendr()` call `use_r_version("4.2")`

**Comment by JosiahParry** [comment]
I'm going to co-opt this issue. When using `use_extendr()` two things should happen: 

1. `usethis::use_version("4.2", "Depends")` to set the minimum version of R
2. Set `rust-version = 1.68.0` to set the MSRV of extendr on the crate
